utils/generate_coverage

```diff
@@ -0,0 +1,56 @@
+#!/usr/bin/env bash
+#
+# Copyright (c) 2021 Seagate Technology LLC and/or its Affiliates
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# For any questions about this software or licensing,
+# please email opensource@seagate.com or cortx-questions@seagate.com.
+
+HARE_BASE_DIR=/opt/seagate/cortx/hare
+PYTHONPATH="$HARE_BASE_DIR/lib/python3.6/site-packages"
+PYTHONPATH="$HARE_BASE_DIR/lib64/python3.6/site-packages:$PYTHONPATH"
+export PYTHONPATH
+		
+PATH="$HARE_BASE_DIR/bin:$PATH"
+export PATH
+
+currentdir_path=`pwd`
+curren_dir_name=${currentdir_path##*/}
+
+check_hare_home() {
+    if [[ $curren_dir_name != 'hare' ]]; then
```

Comment:
I don't think that this check is valid.
In my case the source folder with Hare sources is named `cortx-hare`. 
Why do we need this check? What problem does it solve?

Reply:
to run hare_coverage.py we need to run it from hare home (A parent directory of hax, cfgen, utils and provisioning etc.) so that hax and provisioning source and test code correct path will be get to run the pytest-cov which will create coverage and pytest which run unit testcases.

**pytest Command lines generated through script :**
pytest /root/hare_repo/hare/provisioning/miniprov/test
pytest /root/hare_repo/hare/hax/test
 pytest --cov=hax/hax --cov-report=xml:hare_coverage_report/coverage.xml hax/test/
 pytest --cov=provisioning/miniprov/hare_mp --cov-report=xml:hare_coverage_report/coverage.xml --cov-append provisioning/miniprov/test/ 

If someone executes the commands from other directories than it will fail due to not found the directory.

Reply:
As utils/hare_cov/hare_coverage.py line 138 we are doing same so we can remove this check from here.

Reply:
Please check my comment above. I believe we don't need to require any specific current directory, your script can deduct the absolute path to the tests to be executed automatically, assuming that this script is located somewhere near to the test folder.

Reply:
Done